PurchasesCoreSwift/Purchasing/StoreKitWrapper.swift

```diff
@@ -0,0 +1,147 @@
+//  RCStoreKitWrapper.m
+//  Purchases
+//
+//  Created by RevenueCat.
+//  Copyright © 2019 RevenueCat. All rights reserved.
+//
+import StoreKit
+
+// todo: make internal
+@objc(RCStoreKitWrapperDelegate) public protocol StoreKitWrapperDelegate: AnyObject {
+    @objc func storeKitWrapper(_ storeKitWrapper: StoreKitWrapper,
+                               updatedTransaction transaction: SKPaymentTransaction)
+
+    @objc func storeKitWrapper(_ storeKitWrapper: StoreKitWrapper,
+                               removedTransaction transaction: SKPaymentTransaction)
+
+    @objc(storeKitWrapper:shouldAddStorePayment:forProduct:)
+    func storeKitWrapper(_ storeKitWrapper: StoreKitWrapper,
+                         shouldAddStorePayment payment: SKPayment,
+                         for product: SKProduct) -> Bool
+
+    @objc func storeKitWrapper(_ storeKitWrapper: StoreKitWrapper,
+                               didRevokeEntitlementsForProductIdentifiers productIdentifiers: [String])
+}
+
+// todo: make internal
+@objc(RCStoreKitWrapper) public class StoreKitWrapper: NSObject, SKPaymentTransactionObserver {
+
+    @available(macOS 10.14, macCatalyst 13.0, *)
+    @objc public static var simulatesAskToBuyInSandbox = false
+
+    @objc public weak var delegate: StoreKitWrapperDelegate? {
+        didSet {
+            if delegate != nil {
+                paymentQueue.add(self)
+            } else {
+                paymentQueue.remove(self)
+            }
+        }
+    }
+
+    private var paymentQueue: SKPaymentQueue
+
+    @objc public init(paymentQueue: SKPaymentQueue) {
+        self.paymentQueue = paymentQueue
+        super.init()
```

Comment:
`super.init()` not required when superclass is `boring` (your super class has a single designated initializer with a zero-argument init)